src/OpenTelemetry/Trace/Configuration/SpanProcessorPipelineBuilder.cs

```diff
@@ -0,0 +1,96 @@
+﻿// <copyright file="SpanProcessorPipelineBuilder.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Trace.Export;
+
+namespace OpenTelemetry.Trace.Configuration
+{
+    public class SpanProcessorPipelineBuilder
+    {
+        private Func<SpanExporter, SpanProcessor> lastProcessorFactory;
+        private List<Func<SpanProcessor, SpanProcessor>> processorChain;
+
+        internal SpanProcessorPipelineBuilder()
+        {
+        }
+
+        internal SpanExporter Exporter { get; private set; }
+
+        internal List<SpanProcessor> Processors { get; private set; }
+
+        public SpanProcessorPipelineBuilder AddProcessor(Func<SpanProcessor, SpanProcessor> chain)
+        {
+            if (chain == null)
+            {
+                throw new ArgumentNullException(nameof(chain));
+            }
+
+            if (this.processorChain == null)
+            {
+                this.processorChain = new List<Func<SpanProcessor, SpanProcessor>>();
+            }
+
+            this.processorChain.Add(chain);
+
+            return this;
+        }
+
+        public SpanProcessorPipelineBuilder SetExportingProcessor(Func<SpanExporter, SpanProcessor> export)
```

Comment:
Can you add a summary tag to understand the use of this API

Reply:
Sure I can add docs,  I want to get an agreement is this API is good before satisfying stylecop and tests ;)

What's your opinion on usability (in PR description)?

Reply:
It was hard to understand the PR. Thus I asked for docs. I looked at example and understand it better now. Basically this is special processor that can call exporter